tests/mock_network/churn.rs

```diff
@@ -234,9 +237,38 @@ fn random_churn<R: Rng>(
     assert!(max_drop > 0);
     let dropped_nodes = drop_random_nodes(rng, nodes, Some(max_drop));
     warn!("Dropping nodes: {:?}", dropped_nodes);
+
     BTreeSet::new()
 }
 
+// To ensure the dropped nodes are removed from correspondent section members.
+// Sending out enough user data so that the dropped node can be detected via unresponsive tracking.
+fn verify_dropped_nodes<R: Rng>(
+    rng: &mut R,
+    mut nodes: &mut Vec<TestNode>,
+    dropped_nodes: &BTreeSet<PublicId>,
+) {
+    for _ in 0..UNRESPONSIVE_WINDOW {
+        let event: Vec<_> = rng.gen_iter().take(100).collect();
+        nodes.iter_mut().for_each(|node| {
+            let _ = node
+                .inner
+                .elder_state_mut()
+                .map(|state| state.vote_for_user_event(event.clone()));
+        });
+        poll_and_resend(&mut nodes);
```

Comment:
Seem we want to only do `poll_and_resend` once.
i.e vote for all event we need to vote (i.e UNRESPONSIVE_WINDOW), then call poll_and_resend.
then we should assert that the dropped_node is gone because it should be, i.e panic otherwise, but not loop to try again?

Reply:
hmm...  sending too many user data at once could cause other valid node to be accused as unresponsive, as the `votes` from other peers haven't arrived yet via polling.
That's why it has a `vote -> poll -> check` pattern here.
Maybe I could change it to send out a bunch of Threshold votes first, then vote & poll one by one?

Reply:
Did I miss something, but I thought that would not be an issue since we only track the vote that reached consensus? Why do you think you need to do one at a time? That is not how things will happen in real world, we will have a bunch of votes between each gossip event.

Reply:
we only requires `supermajority` to reach consensus, and the distribution of messages are not random enough. Which means there is chance that some nodes' vote always got received AFTER consensus, and be accused of un-responsive.
It won't happen often, but may fail the soak test with certain chance.